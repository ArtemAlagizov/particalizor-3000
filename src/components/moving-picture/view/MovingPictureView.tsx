import React from "react";
import { Styled } from "../styled";
import { IMovingPictureViewProps } from "./IMovingPictureViewProps";

const MovingPictureView: React.FC<IMovingPictureViewProps> = ({
  destinationCanvasRef,
  referenceCanvasRef
}) => {
  return (
    <>
      <Styled.DestinationCanvas>
        <canvas ref={destinationCanvasRef} className="dest-canvas" />
      </Styled.DestinationCanvas>
      <Styled.RefCanvas>
        <canvas id="referenceCanvas" ref={referenceCanvasRef} />
      </Styled.RefCanvas>
    </>
  );
};

export default MovingPictureView;
