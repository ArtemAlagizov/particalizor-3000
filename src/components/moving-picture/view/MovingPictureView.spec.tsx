import React from "react";
import { shallow } from "enzyme";
import MovingPictureView from "./MovingPictureView";

describe("Given MovingPictureView", () => {
  describe("when rendering", () => {
    it("should have MovingPictureView matching snapshot", () => {
      const component = shallow(
        <MovingPictureView
          imageHeight={0}
          imageWidth={0}
          referenceCanvasRef={null}
          destinationCanvasRef={null}
        />
      );

      expect(component).toMatchSnapshot();
    });
  });
});
