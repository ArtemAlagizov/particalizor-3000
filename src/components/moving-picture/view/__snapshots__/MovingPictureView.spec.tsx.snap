// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given MovingPictureView when rendering should have MovingPictureView matching snapshot 1`] = `
<Fragment>
  <styled.div>
    <canvas
      className="dest-canvas"
    />
  </styled.div>
  <styled.div>
    <canvas
      id="referenceCanvas"
    />
  </styled.div>
</Fragment>
`;
